### PR TITLE
Improve Docker build layout and naming

### PR DESCRIPTION
**Pull request by joshuaboniface**
Flips the script names and adds comments to the various scripts.

This makes the main script `build`, to match the main jellyfin/jellyfin repo and thus enabling consistent building.

Also adds a release spec option to specify the type of build.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,6 +1,7 @@
-.git
+.git*
 .dockerignore
 Dockerfile
-docker-build.sh
+build
+new-file-header.txt
```
[the remark this answers is not in the document]
Comment (joshuaboniface) [reply .dockerignore]: No real reason for it to be copied into the build environment - it makes no real difference, but figured i'd add it on principle.

File: Dockerfile [hunk 2]
```diff
@@ -1,32 +1,42 @@
...
 
+# Install Android repository tools
 RUN rm -rf ${ANDROID_DIR}/tools \
  && wget http://dl-ssl.google.com/android/repository/tools_r25.2.3-linux.zip -O tools.zip \
  && unzip tools.zip -d ${ANDROID_DIR}/ \
  && rm -f tools.zip
 
-# Required to accept licenses:
+# Install SDK tools
+# `yes` is equired to accept licenses:
```
[the remark this answers is not in the document]
Comment (joshuaboniface) [reply Dockerfile]: Fixed.

File: docker-build.sh [hunk 3]
```diff
@@ -1,57 +1,48 @@
...
+# Export environment variables
+export ANDROID_HOME=${ANDROID_DIR}
+export NODE_ENV
+
+# Move to source directory
+pushd ${SOURCE_DIR}
+
+# Install dependencies
+npm cache verify
+sed -i 's/"src/"file:src/g' package.json
```
Comment (JustAMan) [inline docker-build.sh]: is this still needed with latest tools?

Comment (joshuaboniface) [reply docker-build.sh]: For right now, yes, but won't be after Cordova is updated.

**Pull request merged**